File: Learn_DataStructure/unionFind/UF.java
package unionFind;

public class UF {
    private int[] id;
    private int count;
    public UF(int N){
        count=N;
        id=new int[N];
        for(int i=0;i<N;i++)
            id[i]=i;
    }
    public int count(){
        return count;
    }
    public boolean connected(int p,int q){
        return find(p)==find(q);
    }
    public int find(int p){
        return id[p];
    }
    public void union(int p,int q){
        int pID=find(p);
        int qID=find(q);
        if(pID==qID)
            return;
        for(int i=0;i<id.length;i++){
            if(id[i]==pID)
                id[i]=qID;
        }
        count--;
    }
    public void show(){
        for(int i=0;i<id.length;i++){
            System.out.print(id[i]);
        }
        System.out.println();
    }
}

File: Algorithm_Exercises/Algorithm (4th edition)/Algorithm/Algorithm/UF.cs
using System;
namespace Chapter1
{
    public class UF
    {
        private int[] id;
        private int Count { get; set; }
        public UF(int N)
        {
            Count = N;
            id = new int[N];
            for (int i = 0; i < N; i++)
                id[i] = i;
        }

        public bool Connected(int p, int q) => Find(p) == Find(q);

        public int Find(int p)
        {
            while (p != id[p])
                p = id[p];
            return p;
        }

        public void Union(int p, int q)
        {
            int pRoot = Find(p);
            int qRoot = Find(q);
            if (pRoot == qRoot)
                return;
            id[pRoot] = qRoot;
            Count--;
        }
                
    }
}
